extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/PathMatchingHttpSecurityPolicy.java

```diff
@@ -93,7 +93,10 @@ void init(HttpBuildTimeConfig config, Map<String, Supplier<HttpSecurityPolicy>>
                     List<HttpMatcher> perms = new ArrayList<>();
                     tempMap.put(path, perms);
                     perms.add(m);
-                    if (path.endsWith("*")) {
+                    if (path.endsWith("/*")) {
```

Comment:
The original check should be retained as well IMHO, so that the last URL segment paths can be wildcard matched, those not ending with "/"

Reply:
It is retained, it was just moved in to the `else if` branch (line 99)